src/Modules/commentsPopUp.js

```diff
@@ -0,0 +1,147 @@
+const getComments = async (itemId) => {
+  try {
+    const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/kOjkqwEwPrgtqCjTWn0f/comments?item_id=${itemId}`);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+const addComment = async (itemId, name, comment) => {
+  await fetch('https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/kOjkqwEwPrgtqCjTWn0f/comments', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      item_id: itemId,
+      username: name,
+      comment,
+    }),
+  });
+};
+
+const commentsPopUp = () => {
+  const likeBtns = Array.from(document.getElementsByClassName('btn'));
+  const pop = document.querySelector('.pop');
+  pop.setAttribute('style', 'display: none;');
+
+  const createPopup = async (data) => {
+    const popup = document.createElement('div');
+    popup.classList.add('popup');
+
+    const closeBtn = document.createElement('div');
+    closeBtn.classList.add('close-btn', 'text-4xl', 'pb-2', 'cursor-pointer', 'text-right');
+    closeBtn.innerHTML = '&times;';
+    closeBtn.addEventListener('click', () => {
+      popup.remove();
+      pop.setAttribute('style', 'display: none;');
+    });
+
+    const popupImgContainer = document.createElement('div');
+    popupImgContainer.classList.add('img-container', 'rounded-lg');
+
+    const popupImg = document.createElement('img');
+    // popupImg.classList.add('')
+    popupImg.src = data.images?.coverart || data.images?.background || `https://dummyimage.com/400x400/000/1aff00&text=${data.title}`;
+
+    const popupTitle = document.createElement('h3');
+    popupTitle.classList.add('text-3xl', 'font-bold', 'pt-4');
+    popupTitle.innerHTML = data.title;
+
+    const artist = document.createElement('h4');
+    artist.classList.add('text-[#777]', '-mt-2');
+    artist.innerHTML = data.artists[0].alias;
+
+    const commentsTitle = document.createElement('h3');
+    const comments = await getComments(data.key);
+
+    commentsTitle.classList.add('text-[#1ED760]', 'mt-4');
+    commentsTitle.textContent = `Comments (${comments.length ? comments.length : '0'})`;
+
+    const commentList = document.createElement('ul');
+    commentList.classList.add('comment-list');
+    if (comments.length > 0) {
+      comments.forEach((comment) => {
+        const commentItem = document.createElement('li');
+        commentItem.innerHTML = `${comment.creation_date} <strong>${comment.comment} : </strong>  ${comment.username}`;
+        commentList.appendChild(commentItem);
+      });
+    } else {
+      const noCommentItem = document.createElement('li');
+      noCommentItem.innerHTML = 'No comments yet';
+      commentList.appendChild(noCommentItem);
+    }
+
```

Comment:
- [OPTIONAL] The bug that I talked about in the reservation popup is also found in the comments popup. This bug was missed by the previous reviewer, but I will highly recommend you to fix it.
  - When I open a comments popup that has `Comments(0)`, then when I add the first comment, The heading turns `Comments(undefined)` and then When I add the second comment, it turns out to be `Comments(1)` when you can see 2 comments down there. But when I close the popup and reopen it again, the counter gets corrected.
  Kindly consider fixing it.

https://user-images.githubusercontent.com/96683843/235302356-000c3cfb-0f32-441e-99b8-8e44bc9e9a66.mov